File: app/main/coins.lua
local ____exports = {}
function ____exports.is_enough_coins(amount)
    return GameStorage.get("coins") >= amount
end
function ____exports.add_coins(amount)
    local coins = GameStorage.get("coins")
    coins = math.min(coins + amount, 10000)
    GameStorage.set("coins", coins)
    EventBus.send("ADDED_COIN")
end
function ____exports.remove_coins(amount)
    local coins = GameStorage.get("coins")
    coins = coins - amount
    GameStorage.set("coins", coins)
    EventBus.send("REMOVED_COIN")
end
return ____exports

File: src/main/coins.ts
export function is_enough_coins(amount: number) {
    return GameStorage.get('coins') >= amount;
}

export function add_coins(amount: number) {
    let coins = GameStorage.get('coins');
    coins = math.min(coins + amount, 10000);
    GameStorage.set('coins', coins);
    EventBus.send('ADDED_COIN');
}

export function remove_coins(amount: number) {
    let coins = GameStorage.get('coins');
    coins -= amount;
    GameStorage.set('coins', coins);
    EventBus.send('REMOVED_COIN');
}